internal/services/machinelearning/machine_learning_workspace_resource.go

```diff
@@ -230,6 +231,26 @@ func resourceMachineLearningWorkspace() *pluginsdk.Resource {
 				Default:  false,
 			},
 
+			"serverless_compute": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"subnet_id": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: networkValidate.SubnetID,
```

Comment:
We should use the Validation function from the `commonids` package for subnets
```suggestion
							ValidateFunc: commonids.ValidateSubnetID,
```